hnn_core/viz.py

```diff
@@ -314,8 +317,25 @@ def plot_spikes_hist(cell_response, ax=None, spike_types=None, show=True):
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    spike_times = np.array(sum(cell_response._spike_times, []))
-    spike_types_data = np.array(sum(cell_response._spike_types, []))
+    n_trials = len(cell_response.spike_times)
+    if trial_idx is None:
+        trial_idx = list(range(n_trials))
+
+    if isinstance(trial_idx, int):
+        trial_idx = [trial_idx]
+    _validate_type(trial_idx, list, 'trial_idx', 'int, list of int')
+
+    # Extract desired trials
+    if cell_response._spike_times[0]:
```

Comment:
do you mean to check the length > 0 or do you want to compare against `None`?

Reply:
updated!